_posts/2020/2020-03-01-fix-first-last-explicitly-set-implicit-ordering-in-rails-with-uuids.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "‘Fix’ first & last by explicitly setting implicit ordering"
+description: "When UUIDs and Rails make a mess"
+layout: article
+category: ruby
+image:
+  base: '2020/fix-first-last-explicitly-set-implicit-ordering-in-rails-with-uuids'
+  alt: "Pins on a red mat"
+  credit:  Jeff Frenette
+  source: "https://unsplash.com/photos/Y_AWfh0kGT4"
+---
+
+I strongly suggest for new applications, to [use UUIDs as primary keys](/ruby/choose-uuids-for-model-ids-in-rails). However this causes issues in the implicit ordering that Rails performs.
+
+I proposed a method for rectifying that by [using named scopes alongside `first` and `last`](/ruby/first-and-last-may-not-mean-what-you-think), but since that article there is an easier way to re-enable the default behaviour of Active Record.
+
+
+## Instead of...
+
+...avoiding `#first` & `#last` when using none sequential `id`s on an Active Record model. Or using the extra specificity provided by a named scope...
+
+
+## Use
+
+...`implicit_order_column` within your model and aim it at the automatically generated `created_at` column.
+
+```ruby
+class Coffee < ApplicationRecord
+  self.implicit_order_column = "created_at"
+end
+```
+
+
+## Why?
+
+The tradeoff when using UUIDs in your database remains a major benefit: uniqueness, assignability & security.
+
+With this one-line change you also retain the benefits of Rails’ predictability of `.first` and `.last`, provided you assign to the `created_at` timestamp.
+
+My friend [Tekin](https://tekin.co.uk) came up with the idea and [implemented it in Rails](https://github.com/rails/rails/pull/34480)!
```

Comment:
This should be at the bottom. How you normally do acknowledgements.

Breaks the flow here.

Reply:
And now you've said this I think I already knew about this — doesn't Tekin mention in his talk? Or maybe I saw it on Twitter — but I forgot! Thanks for reminding me!